### PR TITLE
disabled-esdt-builtIn

### PR DESCRIPTION
**Pull request by sasurobert**
disabled built in function calls, esdt and governance contracts

### CODE REVIEW

File: vm/systemSmartContracts/governance_test.go [hunk 1]
```diff
@@ -284,6 +284,55 @@ func TestGovernanceContract_ExecuteWhiteListProposalShouldWork(t *testing.T) {
...
+		SetStorageCalled: func(key []byte, value []byte) {
+			if strings.Contains(string(key), string(proposalPrefix)) {
+				genProposal := &GeneralProposal{}
+				_ = json.Unmarshal(value, genProposal)
+				require.Equal(t, gitHubCommit, genProposal.GitHubCommit)
+				require.Equal(t, startNonce, genProposal.StartVoteNonce)
+				require.Equal(t, stopNonce, genProposal.EndVoteNonce)
+				require.Equal(t, callerAddr, genProposal.IssuerAddress)
+				require.False(t, genProposal.Voted)
+
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/governance_test.go:312]: Delete empty line

File: vm/systemSmartContracts/governance_test.go [hunk 2]
```diff
@@ -284,6 +284,55 @@ func TestGovernanceContract_ExecuteWhiteListProposalShouldWork(t *testing.T) {
 	require.Equal(t, vmcommon.Ok, retCode)
 }
 
+func TestGovernanceContract_ExecuteWhiteListProposalShouldWorkDisabled(t *testing.T) {
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/governance_test.go:287]: ShouldNotWorkDisabled ?

File: vm/systemSmartContracts/esdt.go [hunk 3]
```diff
@@ -80,9 +82,16 @@ func (e *esdt) Execute(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
...
+	if args.Function == core.SCDeployInitFunctionName {
 		return e.init(args)
+	}
+
+	if e.disabled {
+		e.eei.AddReturnMessage("ESDT SC disabled")
+		return vmcommon.UserError
+	}
+
+	switch args.Function {
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/esdt.go:94]: In governanceContract in the same method Execute, if no one args.Function is found in switch it is returned in the last line: 

	g.eei.AddReturnMessage("invalid method to call")
	return vmcommon.FunctionNotFound

In this method in line 119 it is returned: vmcommon.Ok

This was the intention?

Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go:94]: thx. fix in another PR as was doing trail of bits audit findings as well.

File: vm/systemSmartContracts/governance_test.go (hunk 1 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance_test.go:312]: done

File: vm/systemSmartContracts/governance_test.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance_test.go:287]: done

File: vm/systemSmartContracts/esdt.go (hunk 3 above)
Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go:94]: fixed here.

File: process/smartContract/process.go [hunk 4]
```diff
@@ -233,7 +233,11 @@ func (sc *scProcessor) doExecuteSmartContractTransaction(
 	executedBuiltIn, err = sc.resolveBuiltInFunctions(txHash, tx, acntSnd, acntDst, vmInput)
 	if err != nil {
 		log.Debug("processed built in functions error", "error", err.Error())
-		return vmcommon.UserError, err
+		errIfError := sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(returnMessage), snapshot)
```
Comment (SebastianMarian) [inline process/smartContract/process.go:239]: errProcessIfError

Comment (sasurobert) [reply process/smartContract/process.go]: refactored

File: cmd/node/config/config.toml [hunk 5]
```diff
@@ -14,6 +14,9 @@
    # DisableDeploy represents the flag to enable/disable deployment of smart contracts
    DisableDeploy = false
 
+   # DisableBuitInFunction represents the flag to enable/disable built in function calls
+   DisableBuitInFunction = false
```
Comment (iulianpascalau) [inline cmd/node/config/config.toml:18]: DisableBuitInFunctions *

Comment (sasurobert) [reply cmd/node/config/config.toml]: done

File: cmd/node/config/config.toml [hunk 6]
```diff
@@ -14,8 +14,8 @@
    # DisableDeploy represents the flag to enable/disable deployment of smart contracts
    DisableDeploy = false
 
-   # DisableBuitInFunction represents the flag to enable/disable built in function calls
-   DisableBuitInFunction = true
+   # DisableBuiltInFunctions represents the flag to enable/disable built in function calls
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:17]: functions calls

File: integrationTests/testProcessorNode.go [hunk 7]
```diff
@@ -1154,6 +1155,7 @@ func (tpn *TestProcessorNode) initMetaInnerProcessors() {
 	}
 	txTypeHandler, _ := coordinator.NewTxTypeHandler(argsTxTypeHandler)
 	tpn.GasHandler, _ = preprocess.NewGasComputation(tpn.EconomicsData, txTypeHandler)
+	badBlocskHandler, _ := tpn.InterimProcContainer.Get(dataBlock.InvalidBlock)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/testProcessorNode.go:1158]: done

File: integrationTests/testProcessorNode.go [hunk 8]
```diff
@@ -1012,6 +1012,7 @@ func (tpn *TestProcessorNode) initInnerProcessors() {
...
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/testProcessorNode.go:1015]: done

File: cmd/node/config/config.toml (hunk 6 above)
Comment (sasurobert) [reply cmd/node/config/config.toml:17]: done

**Pull request merged**